lib/spack/spack/test/installer.py

```diff
@@ -740,6 +741,85 @@ def test_installer_init_requests(install_mockery):
         assert request.pkg.name == spec_name
 
 
+@pytest.mark.parametrize("transitive", [True, False])
+def test_install_spliced(
+    install_mockery, mock_fetch, default_mock_concretization, monkeypatch, capsys, transitive
+):
+    """TODO: description"""
+    spec = default_mock_concretization("splice-t")
+    dep = default_mock_concretization("splice-h+foo")
+
+    # Do the splice.
+    out = spec.splice(dep, transitive)
+    installer = create_installer([out], {"vebose": True, "fail_fast": True})
+    installer.install()
+    for node in out.traverse():
+        assert node.installed
+        assert node.build_spec.installed
+
+
+@pytest.mark.parametrize("transitive", [True, False])
+def test_install_spliced_build_spec_installed(
+    install_mockery, default_mock_concretization, capfd, mock_fetch, transitive
+):
+    """TODO: description"""
+    spec = default_mock_concretization("splice-t")
+    dep = default_mock_concretization("splice-h+foo")
+
+    # Do the splice.
+    out = spec.splice(dep, transitive)
+    out.build_spec.package.do_install()
+    installer = create_installer([out], {"vebose": True, "fail_fast": True})
+    installer._init_queue()
+    for _, task in installer.build_pq:
+        assert isinstance(task, inst.RewireTask if task.pkg.spec.spliced else inst.BuildTask)
+    assert installer.build_pq[-1][0][0] == 2
+    installer.install()
+    for node in out.traverse():
+        assert node.installed
+        assert node.build_spec.installed
+
+
+@pytest.mark.not_on_windows("lacking windows support for binary installs")
+@pytest.mark.parametrize("transitive", [True, False])
+@pytest.mark.parametrize("root_str", ["splice-t^splice-h~foo", "splice-h~foo"])
+def test_install_splice_root_from_binary(
+    install_mockery,
+    default_mock_concretization,
+    mock_fetch,
+    mutable_temporary_mirror,
+    transitive,
+    root_str,
+):
+    """TODO: Docstring"""
+    # Test splicing and rewiring a spec with the same name, different hash.
+    original_spec = spack.spec.Spec(root_str).concretized()
+    spec_to_splice = spack.spec.Spec("splice-h+foo").concretized()
+
+    original_spec.package.do_install()
+    spec_to_splice.package.do_install()
+
+    out = original_spec.splice(spec_to_splice, transitive)
+
+    buildcache = SpackCommand("buildcache")
+    buildcache(
+        "push",
+        "--allow-root",
+        "--unsigned",
+        "--update-index",
+        mutable_temporary_mirror,
+        str(original_spec),
+        str(spec_to_splice),
+    )
```

Comment:
Same issue here with using a `SpackCommand` in a non-command test module.  Unfortunately, not doing so requires a refactor of `cmd/buildcache.py` (though that would be the right thing to do to expose an API for use here).